### PR TITLE
Add cognito users

### PR DESCRIPTION
**Pull request by alvak1**
Access the page `/admin/user-management` and then click on add user button

### CODE REVIEW

File: site/pages/api/add-user.api.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import { addUserSchema } from "../../schemas/add-user.schema";
+import { flattenZodErrors } from "../../utils";
+import {
+    redirectToError,
+    setCookieOnResponseObject,
+    redirectTo,
+    destroyCookieOnResponseObject,
+} from "../../utils/apiUtils";
+
+const addUser = async (req: NextApiRequest, res: NextApiResponse) => {
```
Comment (lj-kpmg) [inline site/pages/api/add-user.api.ts:14]: I've mentioned this in Alaina's PR but I'll put it here as well: We need to restrict access to this api endpoint so only authenticated admins can call it, if we update line 173 in middleware.api.ts to:
```ts
if (request.nextUrl.pathname.startsWith("/admin/") || request.nextUrl.pathname.startsWith("/api/admin/")) {
```
and move this file into an admin folder then that should do the job, the action endpoint for the form will need updating too

File: site/pages/api/add-user.api.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    errors: flattenZodErrors(validatedBody.error),
+                }),
+                res,
+            );
+
+            redirectTo(res, ADD_USER_PAGE_PATH);
+            return;
+        }
+
+        await createUser(validatedBody.data);
```
Comment (lj-kpmg) [inline site/pages/api/add-user.api.ts:31]: We should get the session of the logged in user and make sure that the user they're trying to create is in the same org, otherwise admin users could create other users in any organisation

File: site/pages/admin/user-management.page.tsx [hunk 3]
```diff
@@ -113,7 +113,13 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{ props:
 
     const userList = userManagementSchema
         .parse(userRecords)
-        .filter((user) => user.organisation === sessionWithOrg.orgId);
+        .filter((user) => user.organisation === sessionWithOrg.orgId)
+        .sort((a, b) => {
+            return (
+                a.givenName.toLowerCase().localeCompare(b.givenName.toLowerCase()) ||
```
Comment (lj-kpmg) [inline site/pages/admin/user-management.page.tsx:119]: I would just sort by email here since we don't display first and last names on the page

File: site/pages/admin/add-user.page.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                        widthClass="w"
+                        value={pageState.inputs.email}
+                        initialErrors={pageState.errors}
+                        schema={addUserSchema.shape.email}
+                        stateUpdater={stateUpdater}
+                        maxLength={100}
+                    />
+
+                    <Table rows={[{ header: "Organisation", cells: [pageState.sessionWithOrg?.orgName, ""] }]} />
+                    <input type="hidden" name="orgId" value={pageState.sessionWithOrg?.orgId} />
```
Comment (lj-kpmg) [inline site/pages/admin/add-user.page.tsx:67]: This should be retrieved from the session directly in the API call, otherwise people could pass through whatever orgId they want

File: site/pages/api/add-user.api.ts (hunk 1 above)
Comment (alvak1) [reply site/pages/api/add-user.api.ts]: done

File: site/pages/api/add-user.api.ts (hunk 2 above)
Comment (alvak1) [reply site/pages/api/add-user.api.ts]: done. There is no input for org id. The id is retrieved from the session within the api and passed to cognito call.

File: site/pages/admin/user-management.page.tsx (hunk 3 above)
Comment (alvak1) [reply site/pages/admin/user-management.page.tsx]: done

File: site/pages/admin/add-user.page.tsx (hunk 4 above)
Comment (alvak1) [reply site/pages/admin/add-user.page.tsx]: done. More details in comment above

File: stacks/CognitoStack.ts [hunk 5]
```diff
@@ -28,6 +28,7 @@ export const CognitoStack = ({ stack }: StackContext) => {
         timeout: 30,
         memorySize: 256,
         runtime: "python3.9",
+        enableLiveDev: false,
```
Comment (alainajaffer) [inline stacks/CognitoStack.ts:31]: why is this required?

**Review by lj-kpmg — APPROVED** [review]

File: stacks/CognitoStack.ts (hunk 5 above)
Comment (alvak1) [reply stacks/CognitoStack.ts:31]: The admin create user command was timing out and expected the stack to be up locally
